doc/reference/trd-syscalls.md

```diff
@@ -264,26 +264,51 @@ call to invoke. The kernel invokes callbacks only in response to Yield
 system calls.  This form of very limited preemption allows userspace
 to manage concurrent access to its variables.
 
-There are two Yield system calls:
+There are three Yield system calls:
   - `yield-wait`
   - `yield-no-wait`
+  - `yield-exit`
 
 The first call, `yield-wait`, blocks until a callback executes. This is the
 only blocking system call in Tock. It is commonly used to provide a blocking
 I/O interface to userspace. A userspace library starts a long-running operation
 that has a callback, then calls `yield-wait` to wait for a callback. When the
 `yield-wait` returns, the process checks if the resuming callback was the one
-it was expecting, and if not calls `yield-wait` again. The `yield-wait` system
-call always returns the `Success` variant.
+it was expecting, and if not calls `yield-wait` again. 
 
-The second call, 'yield-no-wait', executes a single callback if any is
-pending.  If no callbacks are pending it returns immediately. The
-`yield-no-wait` system call returns the `Success` variant if a
-callback executed and `Failure` variant if no callback executed.
+The second call, `yield-no-wait`, executes a single callback if any is pending.
+If no callbacks are pending it returns immediately. 
 
-The return variants for Yield system calls are `Success` and
-`Failure`.
+The third call, `yield-exit` never returns and executes no callbacks. It
+causes the process to terminate. The kernel can reclaim any resources held by
+a process after it invokes `yield-exit`.
 
+The register arguments for Subscribe system calls are as follows. The registers
+r0-r3 correspond to r0-r3 on CortexM and a0-a3 on RISC-V.
+
+| Argument               | Register |
+|------------------------|----------|
+| Yield identifer        | r0       |
+| No wait field          | r1       |
+
+
+The yield identifier specifies which call is invoked.
+
+| System call     | Yield identifier value |
+|-----------------|------------------------|
+| yield-no-wait   |                      0 |
+| yield-wait      |                      1 |
```

Comment:
Since `yield-wait` is the common case, I expected that it would be 0, and the unlikely variants would be numbered from there.

Reply:
I thought "0 is false, 1 is true." It also means that 0 does not block, all other values block (some terminally).

Reply:
I think its fine as it is.  It makes more sense to me for it to be structured like this...

0  - no waiting
1 - wait/block until callback
2 - wait/block forever (exit)  

I dont know if one should focus on common case.  It seems the numbering, as it is, correlates with the "waiting" period.